data-prepper-plugin-framework/src/test/java/org/opensearch/dataprepper/plugin/PluginCreatorContextTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugin;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import static org.mockito.Mockito.mock;
+import java.util.Comparator;
+
+public class PluginCreatorContextTest {
+    PluginCreatorContext pluginCreatorContext;
+
+    @BeforeEach
+    void setUp() {
+        pluginCreatorContext = new PluginCreatorContext();
+    }
+    
+    @Test
+    public void test_observablePluginCreator() {
+        PluginCreator pluginCreator = pluginCreatorContext.observablePluginCreator();
+        assertNotNull(pluginCreator);
+    }
+
+    @Test
+    public void test_pluginCreator() {
+        PluginConfigurationObservableRegister pluginConfigurationObservableRegister = mock(PluginConfigurationObservableRegister.class);
+        PluginCreator pluginCreator = pluginCreatorContext.pluginCreator(pluginConfigurationObservableRegister);
+        assertNotNull(pluginCreator);
+    }
+
+    @Test
+    public void test_extensionsLoaderComparator() {
+        Comparator<ExtensionLoader.ExtensionPluginWithContext> extensionsLoaderComparator = pluginCreatorContext.extensionsLoaderComparator();
+        assertNotNull(extensionsLoaderComparator);
```

Comment:
We should include tests that verify the sorting here. That way we can verify different sort combinations.

You don't have to test using sorts per se.

It could be as simple as these some of these combinations on `compareTo`:

* Two objects both with isConfigured==true should have compareTo==0
* Two objects both with isConfigured==false should have compareTo==0
* Two objects, one with isConfigured==false and isConfigured==true, compareTo < 0 when true is on the left
* Two objects, one with isConfigured==false and isConfigured==true, compareTo > 0 when true is on the right

Reply:
It would be better to put these as different `@Test` cases, but I'm ok with it.